modules/packages/Sort.chpl

```diff
@@ -835,14 +837,37 @@ record ReverseComparator {
   var comparator;
 
   /*
-   Constructor - builds a comparator with a compare method that reverses the sort order of
-   the argument-provided comparator.
+   Initializer - builds a comparator with a compare method that
+   reverses the sort order of the default comparator.
+   */
+  proc init() {
+    this.comparator = defaultComparator;
+    super.init();
+  }
+
+  /*
+   Initializer - builds a comparator with a compare method that
+   reverses the sort order of the argument-provided comparator.
 
    :arg comparator: :ref:`Comparator <comparators>` record that defines how the
       data is sorted.
 
    */
-  proc ReverseComparator(comparator:?rec=defaultComparator) {}
+  proc init(comparator) {
+    this.comparator = comparator;
+    super.init();
+  }
+
+  /*
+   Copy Initializer - builds a comparator that's a copy of
+   its argument.
+
+   :arg revcomp: :ref:`ReverseComparator <reverse-comparator>` to copy.
+   */
+  proc init(revcomp: ReverseComparator(?)) {
```

Comment:
I wasn't sure either.  It's your module, so your call.  While users probably won't call it, I wondered whether they'd be happy to simply know that it exists (i.e., that something reasonable would happen if their ReverseComparator was copied).

Reply:
I can't imagine a scenario where this would be needed, and the record-comparators already feel a bit bloated as opposed to first class functions as comparators, so I'm inclined to leave the copy constructor undocumented.